### PR TITLE
bpo-45156: fixes inifite loop on `mock.seal()`

### PR DESCRIPTION
**Pull request by sobolevn**
https://bugs.python.org/issue45156

### CODE REVIEW

**Comment by tirkarthi** [comment]
cc @mariocj89

File: Lib/unittest/mock.py [hunk 1]
```diff
@@ -1005,6 +1005,11 @@ def _get_child_mock(self, /, **kw):
         if _new_name in self.__dict__['_spec_asyncs']:
             return AsyncMock(**kw)
 
+        if self._mock_sealed:
+            attribute = "." + kw["name"] if "name" in kw else "()"
```
Comment (corona10) [inline Lib/unittest/mock.py]: nit
```suggestion
            attribute = f".{kw['name']} if "name" in kw else "()"
```

File: Misc/NEWS.d/next/Tests/2021-09-13-00-28-17.bpo-45156.8oomV3.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Fixes infinite loop on ``unittest.seal()`` of mocks created by
```
Comment (corona10) [inline Misc/NEWS.d/next/Tests/2021-09-13-00-28-17.bpo-45156.8oomV3.rst]:
```suggestion
Fixes infinite loop on :func:`unittest.mock.seal` of mocks created by
```

File: Misc/NEWS.d/next/Tests/2021-09-13-00-28-17.bpo-45156.8oomV3.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
...
+``mock.create_autospec()``.
```
Comment (corona10) [inline Misc/NEWS.d/next/Tests/2021-09-13-00-28-17.bpo-45156.8oomV3.rst]: nit
```suggestion
:func:`~unittest. create_autospec `.
```

**Comment by sobolevn** [comment]
@corona10 fixed! Thanks a lot for your help!

File: Lib/unittest/mock.py (hunk 1 above)
Comment (cjw296) [reply Lib/unittest/mock.py]: Not sure this is better, why do you feel the f-string makes things better here?

**Review by mariocj89 — APPROVED** [review]
Thanks for submitting this!

**Comment by vstinner** [comment]
Thanks @sobolevn for the fix. I trust @mariocj89 for the review, so I merged the PR ;-)

**Comment by sobolevn** [comment]
Thanks everyone!

**Pull request merged**